doc/help/kdb-gen-highlevel.md

```diff
@@ -0,0 +1,101 @@
+
+The template produces two output files: `<outputName>.c` and `<outputName>.h`. The `.c` file only contains implementations,
+therefore its precise content will not be documented.
+
+The header (`.h`) file contains the following parts:
+
+1. Generated `enum`s and `struct`s
+2. Declarations for generated accessor functions
+3. Tags for accessing keys
+4. `static inline` accessor functions for all tags
+5. Declarations of initialization functions
+6. Convenience accessor macros
+
+Anything else that may be part of the header file is not considered public API and may be subject to change at any point in time.
+There is also no guarantee of full compatibility between Elektra version for the documented parts of the header, however,
+we try to have as little breaking changes as possible for the six parts listed above.
+
+For detailed information about the contents of the header file see [elektra-highlevel-gen(7)](elektra-highlevel-gen.md).
+
+## PARAMETERS
+
+- `initFn`:
+  Changes the name of the initialization function (default: `loadConfiguration`)
+- `helpFn`:
+  Changes the name of the function that prints the generated help message (default: `printHelpMessage`)
+- `specloadFn`:
+  Changes the name of the function that checks for "specload mode" (default: `specloadCheck`)
+- `tagPrefix`:
+  Changes the prefix of the generated tags (default: `ELEKTRA_TAG_`)
+- `enumConv`:
+  Switches how enum conversion should be done; allowed values: `default` (default), `trie`, `strcmp`
+  - `strcmp`: uses a simple series of `if (strcmp(*, *) == 0)` to convert strings into enums
+  - `trie`: constructs a character based trie to convert strings into enums
+  - `default`: uses a `trie` up to a depth of 2, then switches to `strcmp`
+- `headers`:
+  Comma-separated (`,`) list of additional header files to include. For each of the listed headers we will generate an `#include "*"`
+  statement
+- `genSetters`:
+  Switches whether setters should be generated at all; allowed values: `true` (default), `false`
+- `specLocation`:
+  Changes where the processed specification will be stored; allowed values: `embedded` (default), `external`.
+  see [elektra-highlevel-gen(7)](elektra-highlevel-gen.md)
+- `defaultsHandling`:
+  Changes how the default values are embedded into the application; allowed values: `embedded` (default), `speconly`.
+  see [elektra-highlevel-gen(7)](elektra-highlevel-gen.md)
+- `specValidation`:
+  Changes how the specification will be validated on application start-up; allowed values: `none` (default), `minimal`.
+  see [elektra-highlevel-gen(7)](elektra-highlevel-gen.md)
+
+## EXAMPLES
+
+The simplest invocation is:
+
+`kdb gen highlevel /sw/org/app/#0/current config`
+
+However, it is not recommended to have the code-generator read from the KDB, so one should instead use:
+
+`kdb gen -F ni=spec.ini highlevel /sw/org/app/#0/current config`
```

Comment:
I tried to execute that (or to be more specific: `kdb gen -F ni=./examples/highlevel/spec.ini highleve /sw/org/app/#0/current config`) I get the error:

```
The command kdb gen terminated unsuccessfully with the info:
parentKey has to be in spec namespace
```

Actually, I think the command is correct, and the spec can simply be appended internally.


And quite weird: I get no (error) output when I mistype "highlevel".

When I strace the command, I get:
```
openat(AT_FDCWD, "./examples/highlevel/spec.ini", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=233, ...}) = 0
read(3, "[]\nmountpoint = highlevelexample"..., 4096) = 233
read(3, "", 4096)                       = 0
close(3)                                = 0
munmap(0x7f83a5a2b000, 2130336)         = 0
exit_group(0)                           = ?
+++ exited with 0 +++
```

So it seems like to successfully open the file, but does not do anything. It would be good to get errors if I did something wrong.

When I change the command with a key starting in spec namespace, I get a config.c and config.h

In the generated code, most of the Elektra in the comments are written incorrectly "elektra".

I'll continue with the review tomorrow.